app/reports/mi-report/create.js

```diff
@@ -0,0 +1,23 @@
+const { writeFile } = require('../../storage')
+const { reportsConfig } = require('../../config')
+const { getEvents } = require('./get-events')
+const { groupEventsByCorrelationId } = require('./group-events-by-correlation-id')
+const { orderGroupedEvents } = require('./order-grouped-events')
+const { getReportLines } = require('./get-report-lines')
+const { convertToCSV } = require('./convert-to-csv')
+
+const createMIReport = async () => {
+  const events = await getEvents()
+  const groupedEvents = groupEventsByCorrelationId(events)
+  const orderedEvents = orderGroupedEvents(groupedEvents)
+  const reportLines = getReportLines(orderedEvents)
+  if (reportLines.length) {
```

Comment:
Is this just checking that the report has content?

Reply:
Yes, if no report lines, then no data so no point creating an empty file.

It mirrors the current Azure Function that doesn't create an empty file either so didn't want to change that behaviour.